### DISCUSSION TITLE
[TS] Transform custom data to match 'messages' property

### DISCUSSION THREAD

**Discussion by LoLo2207** [kazupon/vue-i18n#1629] [Q&A]
Hi, I'm new to TS+Vue+VueI18n and at work we're trying to start a new project with this technologies.

To play around we're getting the translations strings from an API and I need to transform them to match the messages property in createI18n. Coming from a C# background and being new in Typescript, I can't see how would I create an array of these messages to pass to that property.

Currently my translations strings are a `Map<string, Map<string, string>>` where the strings are the locale, the label key and the label value in that locale.

Any help is appreciated. Thanks.

**Comment by kazupon** [comment]
That's best practice to simply use translate strings with **object**.
vue-i18n manages to translate strings by retrieving them from objects.

**Reply by LoLo2207** [reply]
Sorry but I don't understand what you're trying to say. Could I get an example? Thank you.
